packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -104,18 +105,26 @@ export function maybeStripNodeJsVirtualPrefix(
 }
 
 /**
- * Convert any virtual module Id that was generated by the aliases returned from `getNodeCompatAliases()`
- * back to real a module Id and whether it is an external (built-in) package or not.
+ * Resolve the source import to an absolute path, potentially via its alias.
  */
-export function dealiasVirtualNodeJSImport(source: string) {
+export function resolveNodeJSImport(source: string) {
 	const alias = env.alias[source];
-	assert(
-		alias,
-		`Expected "${source}" to have a Node.js compat alias, but none was found`
-	);
-	assert(
-		!env.external.includes(alias),
-		`Unexpected unenv alias to external module: ${source} -> ${alias}`
-	);
-	return alias;
+	if (alias) {
```

Comment:
So, if we can't find the `alias`, we use the `source` as is. In what scenario would this happen?

Reply:
Primarily, the case where the id was on that was injected into the code (to support globals).

Reply:
Don't quite follow. Aren't the injected IDs (in `injectGlobalCode`) always prefixed with `CLOUDFLARE_VIRTUAL_PREFIX`? In that case if they aren't an `alias` then they would be an `external` and the assertion would fail.

Reply:
The injected id are of the form: `@cloudflare/unenv-preset/runtime/node/console/index` and similar.
These are not aliased in the preset's alias, they are just files that need to be injected into the code.
So the transformed code will look like:

```ts
import var_console from "\0__CLOUDFLARE_NODEJS_COMPAT__@cloudflare/unenv-preset/runtime/node/console/index";
globalThis.console = var_console;
```

Reply:
Ah, OK. This is making more sense to me now. Thanks for explaining.

Reply:
```suggestion
	// If `alias` is `undefined` then `source` was injected in the `transform` hook and we can resolve it directly.
	// Else we assign `alias` to `source` before resolving
	if (alias) {
```

Reply:
Added the comments